Src/Plugins/Security/SqlConnectionStringValidator.cs

```diff
@@ -54,9 +63,29 @@ public static string IsValidDynamic(ref string fingerprint, ref string message)
                                                       ref string fingerprintText,
                                                       ref string message)
         {
-            if (!groups.TryGetValue("host", out string host) ||
-                !groups.TryGetValue("account", out string account) ||
-                !groups.TryGetValue("password", out string password))
+            matchedPattern = matchedPattern.Trim();
+
+            string host, database, account, password;
+
+            if (groups.ContainsKey("host") && groups.ContainsKey("database") && groups.ContainsKey("account") && groups.ContainsKey("password"))
+            {
+                host = groups["host"];
+                database = groups["database"];
+                account = groups["account"];
+                password = groups["password"];
+            }
+            else
+            {
+                host = ParseExpression(RegexEngine, matchedPattern, HostExpression);
+                database = ParseExpression(RegexEngine, matchedPattern, DatabaseExpression);
+                account = ParseExpression(RegexEngine, matchedPattern, AccountExpression);
+                password = ParseExpression(RegexEngine, matchedPattern, PasswordExpression);
+            }
+
+            if (string.IsNullOrEmpty(host)
```

Comment:
>IsNullOrEmpty [](start = 23, length = 13)

Can you make these "IsNullOrWhitespace" ?  IMO there's pretty much no reason to use IsNullOrEmpty